src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs

```diff
@@ -187,27 +189,31 @@ private void CopyToCurrentCache(List<ManifestItem> increItems)
         {
             using (new PerformanceScope("CopyToCurrentCache"))
             {
-                foreach (var item in from mi in increItems
-                                     from oi in mi.OutputFiles.Values
-                                     where oi.LinkToPath != null && oi.LinkToPath.StartsWith(_increContext.LastBaseDir)
-                                     select oi)
-                {
-                    string cachedFileName;
-                    if (!_increContext.LastInfo.PostProcessOutputs.TryGetValue(item.RelativePath, out cachedFileName))
+                var itemsToBeCopied = from mi in increItems
+                                      from oi in mi.OutputFiles.Values
+                                      where oi.LinkToPath != null && oi.LinkToPath.StartsWith(_increContext.LastBaseDir)
+                                      select oi;
+                Parallel.ForEach(
+                    itemsToBeCopied,
+                    new ParallelOptions { MaxDegreeOfParallelism = Environment.ProcessorCount },
+                    item =>
                     {
-                        throw new BuildCacheException($"Last incremental post processor outputs should contain {item.RelativePath}.");
-                    }
+                        string cachedFileName;
+                        if (!_increContext.LastInfo.PostProcessOutputs.TryGetValue(item.RelativePath, out cachedFileName))
```

Comment:
at same time, will only get from dictionary without add or update, I think it's safe at this point.

---
In reply to: [107859187](https://github.com/dotnet/docfx/pull/1481#discussion_r107859187) [](ancestors = 107859187)